generator/src/main/java/com/scottlogic/deg/generator/ViolationGenerationEngine.java

```diff
@@ -20,18 +20,18 @@
 import java.util.stream.Collectors;
 
 public class ViolationGenerationEngine implements GenerationEngine {
-    private final GenerationEngine generationEngine;
+    private final StandardGenerationEngine standardGenerationEngine;
     private final Path outputPath;
     private final ManifestWriter manifestWriter;
 
     @Inject
-    public ViolationGenerationEngine(@Named("outputPath") Path outputPath, @Named("valid") GenerationEngine generationEngine, ManifestWriter manifestWriter){
+    public ViolationGenerationEngine(@Named("outputPath") Path outputPath, StandardGenerationEngine standardGenerationEngine, ManifestWriter manifestWriter){
```

Comment:
Same as comment above, implicit vs explicit?

Reply:
For this specifically, you never want the violationGenerationEngine passed in as a dependency to itself.
this change ensures that

Reply:
So the ViolationGenerationEngine is itself a GenerationEngine. The implication here is that it could be used to create itself but what we _actually_ want is an instance of the valid generator, the StandardGenerationEngine.

Reply:
^ didn't see Paul's reply when I was typing